src/System.Runtime.Analyzers/UnitTests/InitializeValueTypeStaticFieldsInlineTests.cs

```diff
@@ -12,12 +12,328 @@ public class InitializeValueTypeStaticFieldsInlineTests : DiagnosticAnalyzerTest
     {
         protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer()
         {
-            return new BasicInitializeValueTypeStaticFieldsInlineAnalyzer();
+            return new BasicInitializeStaticFieldsInlineAnalyzer();
         }
 
         protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
         {
-            return new CSharpInitializeValueTypeStaticFieldsInlineAnalyzer();
+            return new CSharpInitializeStaticFieldsInlineAnalyzer();
         }
+
+        [Fact]
+        public void CSharp_CA1810_NoDiagnostic_EmptyStaticConstructor()
+        {
+            VerifyCSharp(@"
+public class Class1
+{
+    private readonly static int field = 1;
+    static Class1() // Empty
+    {
+    }
+}
+");
+        }
+
+        [Fact]
+        public void CSharp_CA2207_NoDiagnostic_EmptyStaticConstructor()
+        {
+            VerifyCSharp(@"
+public struct Struct1
+{
+    private readonly static int field = 1;
+    static Struct1() // Empty
+    {
+    }
+}
+");
+        }
+
+        [Fact]
+        public void CSharp_CA1810_NoDiagnostic_NoStaticFieldInitializedInStaticConstructor()
```

Comment:
I actually like this. It sorts nicely by language and by diagnostic ID, and it separates positive from negative tests. As fond as I am of "behavioral naming" ("it should do so-and-so"), I think this convention works better for analyzers. Because for analyzers the behavior will always be either "it emits a diagnostic" or "it doesn't".